cupyx/scipy/sparse/base.py

```diff
@@ -292,9 +292,9 @@ def count_nonzero(self):
         """Number of non-zero entries, equivalent to"""
         raise NotImplementedError
 
-    def diagonal(self):
+    def diagonal(self, k=0):
         """Returns the main diagonal of the matrix"""
-        return self.tocsr().diagonal()
+        return self.tocsr().diagonal(k=k)
```

Comment:
Please, change the docstring to reflect that is the k-th diagonal :)
Thanks!

Reply:
done